dll/cpl/sysdm/general.c

```diff
@@ -216,21 +217,20 @@ LRESULT CALLBACK RosImageProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam
             {
                 RECT rcCredits;
                 HDC hDC = GetDC(hwnd);
-                if (hDC == NULL)
-                    break;
-
-                GetClientRect(hwnd, &rcCredits);
-                SetRect(&rcCredits, 0, 0, rcCredits.right, pImgInfo->cySource);
-                FillRect(hDC, &rcCredits, GetSysColorBrush(COLOR_3DFACE));
-
+                if (hDC != NULL)
+                {
+                    GetClientRect(hwnd, &rcCredits);
+                    SetRect(&rcCredits, 0, 0, rcCredits.right, pImgInfo->cySource);
+                    FillRect(hDC, &rcCredits, GetSysColorBrush(COLOR_3DFACE));
+                    ReleaseDC(hwnd, hDC);
+                }
                 KillTimer(hwnd, timerid);
                 DeleteObject(hCreditsBitmap);
-                InvalidateRect(hwnd, NULL, FALSE);
-                ReleaseDC(hwnd, hDC);
 
                 top = 0;
                 timerid = 0;
             }
+            InvalidateRect(hwnd, NULL, FALSE);
```

Comment:
EXPLAIN THIS CHANGE OF BEHAVIOR!

Reply:
Arranged this way for consistency between WM_LBUTTONDOWN and WM_TIMER while the design principle of WM_TIMER was indeed "more correct" (robustness) with regard to top & timerid management as well as positionning of the InvalidRect positionning in the code sequence.

Reply:
Threfore no change required.